### DISCUSSION TITLE
14/5000  How to detect whether the PLC server is down

### DISCUSSION THREAD

**Discussion by javaHBZ** [eclipse-milo/milo#919] [General]
I have a function that needs to detect if the server is down. If it is down, I need to update some data to the offline state

**Comment by javaHBZ** [comment]
I found a `KeepAliveFailure` class in the session package, but I don't know if it is

**Comment by kevinherron** [comment]
Easiest is probably add a `SessionActivityListener` via `OpcUaClient::addSessionActivityListener`. You will get notified when the session goes active or inactive, which more or less corresponds to being connected or not, since you can't really do anything without an active session.
